cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -222,7 +222,9 @@ func (o *ScaleUpOrchestrator) ScaleUp(
 			return buildNoOptionsAvailableStatus(markedEquivalenceGroups, skippedNodeGroups, nodeGroups), nil
 		}
 		var scaleUpStatus *status.ScaleUpStatus
-		createNodeGroupResults, scaleUpStatus, aErr = o.CreateNodeGroup(bestOption, nodeInfos, schedulablePodGroups, podEquivalenceGroups, daemonSets, allOrNothing)
+		oldId := bestOption.NodeGroup.Id()
+		initializer := NewAsyncNodeGroupInitializer(bestOption.NodeGroup, nodeInfos[oldId], o.scaleUpExecutor, o.taintConfig, daemonSets, o.processors.ScaleUpStatusProcessor, o.autoscalingContext, allOrNothing)
```

Comment:
Creation of the initializer used to be flag guarded and here it is no longer the case - is that intentional? If not, can you keep the flag guard?

Reply:
It may not be ideal, but I preferred this over the alternatives:
- passing around a `nil`
- creating a dummy initializer implementation for the case when flag is not flipped
Overall creation of the initializer doesn't really do anything yet.

One obvious option that might make more sense (PLMK WDYT) is to split off orchestrator's `CreateNodeGroupAsync` method.

Reply:
Can you elaborate a bit on why you rejected passing around a `nil`? `CreateNodeGroup` could be doing a nil check instead of `o.autoscalingContext.AsyncNodeGroupsEnabled` check. Right now there's not a lot of logic in `NewAsyncNodeGroupInitializer`, but still it is code we execute even when `o.autoscalingContext.AsyncNodeGroupsEnabled` is false.